### PR TITLE
[JEX-179] Support setting package type for Artifactory 4 repositories

### PR DESCRIPTION
**Pull request by yzl**
Mostly @antoniobeyah's work.  I set a different default package type and did some work to get the tests working again.

cc @chef/engineering-services

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -5,4 +5,5 @@ group :test do
   gem 'sinatra', '~> 1.4'
   gem 'rspec',   '~> 3.0'
   gem 'webmock', '~> 1.17'
+  gem 'rspec-mocks', '3.4.0'
```
Comment (sersut) [inline Gemfile]: Let's add an inline comment here about why are we doing this.

**Comment by sersut** [comment]
👍 other than a minor comment. 

I was not able to find how `attributes` of a `Repository` resource being consumed 😄 But I think the change makes sense.

File: Gemfile (hunk 1 above)
Comment (yzl) [inline Gemfile]: Done, and squashed.

**Comment by schisamo** [comment]
👍  Looks good @yzl!

@sersut Most of the core logic for resources is in `Artifactory::Resource::Base`. In particular the `attribute` method defines a method named _after_ the attribute:
https://github.com/chef/artifactory-client/blob/0e2fe203608ee3f62fc86c404a590f0cbe6fff30/lib/artifactory/resources/base.rb#L24-L71

This is then called in various places when marshalling and unmarshalling JSON/XML received/sent to the Artifactory endpoints:
https://github.com/chef/artifactory-client/blob/0e2fe203608ee3f62fc86c404a590f0cbe6fff30/lib/artifactory/resources/base.rb#L167-L193
https://github.com/chef/artifactory-client/blob/0e2fe203608ee3f62fc86c404a590f0cbe6fff30/lib/artifactory/resources/base.rb#L298-L325

Hopefully all that makes sense.

**Comment by scotthain** [comment]
For form and content 👍 - can't speak specifically to any functionality

**Pull request merged**